packages/salesforcedx-utils/src/cli/checkCliVersion.ts

```diff
@@ -0,0 +1,125 @@
+        new SfdxCommandBuilder().withArg('--version').withJson().build(),
+        {}
+      ).execute();
+      // Save the result of the command
+      const sfdxCmdOutput = new CommandOutput();
+      const sfdxVersion = await sfdxCmdOutput.getCmdResult(sfdxExecution);
+      return sfdxVersion;
+    } catch {
+      return 'No SFDX CLI';
+    }
+  }
+
+  public async getSfCliVersion(): Promise<string> {
+    try {
+      // Execute the command "sf --version" in the Terminal
+      const sfExecution = new CliCommandExecutor(
+        new SfCommandBuilder().withArg('--version').withJson().build(),
+        {}
+      ).execute();
+      // Save the result of the command
+      const sfCmdOutput = new CommandOutput();
+      const sfVersion = await sfCmdOutput.getCmdResult(sfExecution);
+      return sfVersion;
+    } catch {
+      return 'No SF CLI';
+    }
+  }
+
+  public async parseSfdxCliVersion(sfdxCliVersion: string): Promise<number[]> {
+    // Both SFDX and SF v2 can match a valid SFDX version, so there are 2 patterns to search for
+    const sfdxPattern = /sfdx-cli\/(\d+\.\d+\.\d+)/;
+    const sfdxMatch = sfdxPattern.exec(sfdxCliVersion);
+    const sfPattern = /@salesforce\/cli\/(\d+\.\d+\.\d+)/;
+    const sfMatch = sfPattern.exec(sfdxCliVersion);
+    if (sfdxMatch) {
+      const sfdxVersion = sfdxMatch[1];
+      const sfdxVersionNumber = sfdxVersion.split('.').map(Number);
+      return sfdxVersionNumber;
+    }
+    else if (sfMatch) {
+      const sfdxVersion = sfMatch[1];
+      const sfdxVersionNumber = sfdxVersion.split('.').map(Number);
+      return sfdxVersionNumber;
+    }
+    return [-1];
+  }
+
+  public async parseSfCliVersion(sfCliVersion: string): Promise<number[]> {
+    const sfPattern = /@salesforce\/cli\/(\d+\.\d+\.\d+)/;
+    const sfMatch = sfPattern.exec(sfCliVersion);
+    if (sfMatch) {
+      const sfVersion = sfMatch[1];
+      const sfVersionNumber = sfVersion.split('.').map(Number);
+      return sfVersionNumber;
+    }
+    return [-1];
+  }
+
+  public async validateCliInstallationAndVersion(sfdxCliVersionArray: number[], sfCliVersionArray: number[]): Promise<Number> {
+    // The last working version of SFDX is v7.193.2
+    const minSFDXVersion = '7.193.2';
+    const minSFDXVersionArray = minSFDXVersion.split('.').map(Number);
+
+    // SF v1 does not map sf commands to sfdx commands and is not supported
+    const minSFVersion = '2.0.0';
+    const minSFVersionArray = minSFVersion.split('.').map(Number);
+
+    // Case 1: Neither SFDX CLI nor SF CLI is installed
+    if (sfdxCliVersionArray[0] === -1 && sfCliVersionArray[0] === -1) {
+      return CheckCliEnum.cliNotInstalled;
+    }
+
+    // Case 2: Only SF CLI (v1) is installed (SF v1 cannot be used because it does not map sf to sfdx)
+    if (sfdxCliVersionArray[0] === -1 && sfCliVersionArray[0] === 1) {
+      return CheckCliEnum.onlySFv1;
+    }
+
+    // Case 3: Both SFDX CLI (v7) and SF CLI (v2) are installed at the same time
```

Comment:
cant this be simplified to following cases:
1. sfdx and sf both not installed
2. sf v1 installed
3. if sfdx installed, check if sfdx version is supported
4. if sf v2 is installed, check sf version is supported

Reply:
Resolved during our 1:1 on Monday 12/4.

Summary of the discussion: Case 3 (Both SFDX CLI (v7) and SF CLI (v2) are installed at the same time) is necessary for semantic reasons.  If the user has an outdated version of SFDX v7 that they installed simultaneously with a valid version of SF v2, the VSCode Terminal prefers the outdated version of SFDX v7, even though it is incompatible with the extensions.  If we remove Case 3, the user will see the error message that they have an outdated version of SFDX but get confused when they log a Github issue and tell us that they have SF v2 yet the extensions are not working.  This extra check for semantics is to remove that confusion by giving a clear error message to the user about what is happening with their system.